ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveRelColumnsAlignment.java

```diff
@@ -119,9 +119,11 @@ public RelNode align(Aggregate rel, List<RelFieldCollation> collations) {
     final RelNode child = dispatchAlign(rel.getInput(), propagateCollations.build());
 
     // 3) We annotate the Aggregate operator with this info
-    final HiveAggregate newAggregate = (HiveAggregate) rel.copy(rel.getTraitSet(),
```

Comment:
No need to change this, in fact this logic does not need to run for operators pushed to JDBC. Instead, in method
```public RelNode align(RelNode rel, List<RelFieldCollation> collations)``` below (the general case), check if rel is instance of ```HiveJdbcConverter``` and just return the node itself in that case.

Reply:
did it for dispatchAlign (line 92)